drive/src/drive/query.rs

```diff
@@ -51,23 +54,37 @@ impl Drive {
         query_cbor: &[u8],
         contract_id: [u8; 32],
         document_type_name: &str,
+        epoch: Option<&Epoch>,
         transaction: TransactionArg,
     ) -> Result<(Vec<Vec<u8>>, u16, u64), Error> {
         let mut drive_operations: Vec<DriveOperation> = vec![];
         let contract = self
-            .get_contract(contract_id, transaction, &mut drive_operations)?
+            .get_contract_with_fetch_info_and_add_to_operations(
+                contract_id,
+                epoch,
+                transaction,
+                &mut drive_operations,
+            )?
             .ok_or(Error::Query(QueryError::ContractNotFound(
                 "contract not found",
             )))?;
-        let document_type = contract.document_type_for_name(document_type_name)?;
+        let document_type = contract
+            .contract
+            .document_type_for_name(document_type_name)?;
         let (items, skipped) = self.query_documents_from_contract_internal(
-            &contract,
+            &contract.contract,
             document_type,
             query_cbor,
             transaction,
             &mut drive_operations,
         )?;
-        let (_, cost) = calculate_fee(None, Some(drive_operations))?;
+        let cost = if let Some(epoch) = epoch {
```

Comment:
Does `Option` has any elegant monads for this case?

Reply:
You could do a map and then an unwrap_or_default, but I don't think that it's really more elegant.